glib/src/commonMain/kotlin/io/github/cgr/glib/LazyList.kt

```diff
@@ -0,0 +1,162 @@
+package io.github.cgr.glib
+
+class LazyList(private val generator: Iterator<Any?>) : List<Any?> {
```

Comment:
How about making it a `LazyList<T>` and letting users use `LazyList<Any>` if they want? In case someone makes a statically typed golflang.

Reply:
But then the operators will have a bunch of ugly unchecked casts...

Reply:
Which ones are those?  Can you make them accept `LazyList<Any?>`s? That'll avoid the unchecked casts

Reply:
I mean in their implementations. You'd have to do a lot of `is LazyList<Any?>` which triggers unchecked cast warnings. This I found out while trying to use generic lazy lists in Myxal

Reply:
I still don't understand, where would you have to do this? And could `is LazyList<*>` avoid the warnings?

Reply:
You'd have to do it when implementing the actual operators. Using a star projection precludes from accessing any elements

Reply:
Well, if you think it's best from experience, I won't stop you. Plus, it doesn't seem too hard to go back and change it if need be.

Reply:
@ysthakur no further comment? im not against adding generics per se, just using them will be cumbersome

Reply:
I agree with bbrk24. I'd like them but we can always add them later if it turns out we really need them